### PR TITLE
DRY hasBitcoinAddresses and window size breakpoints

### PR DESCRIPTION
**Pull request by sisou**
Dedupe the many computations of whether an account has Bitcoin enabled/available; plus deduping window size breakpoint computations.

### CODE REVIEW

**Comment by onmax** [comment]
Another DRY that I have seen is `width.value <= 700` to test if is mobile, maybe we could add `isMobile` or `isSmallScreen` flag in `useWindowSize` 🤔

**Review by onmax — APPROVED** [review]

**Comment by sisou** [comment]
Great idea @onmax! I pushed another commit to do that! :+1:

File: src/components/BtcTransactionList.vue [hunk 1]
```diff
@@ -150,11 +150,8 @@ export default defineComponent({
         const scrollerBuffer = 300;
 
         // Height of items in pixel
-        const { width: windowWidth } = useWindowSize();
-        const itemSize = computed(() => windowWidth.value > 700 // Full mobile breakpoint
-            ? 72
-            : 68, // 64px + 4px margin between items
-        );
+        const { isMobile } = useWindowSize();
+        const itemSize = computed(() => !isMobile.value ? 72 : 68); // 64px + 4px margin between items
```
Comment (danimoh) [inline src/components/BtcTransactionList.vue]:
```suggestion
        const itemSize = computed(() => isMobile.value ? 68 : 72); // mobile: 64px + 4px margin between items
```

File: src/components/TransactionList.vue [hunk 2]
```diff
@@ -163,11 +163,8 @@ export default defineComponent({
         const scrollerBuffer = 300;
 
         // Height of items in pixel
-        const { width: windowWidth } = useWindowSize();
-        const itemSize = computed(() => windowWidth.value > 700 // Full mobile breakpoint
-            ? 72
-            : 68, // 64px + 4px margin between items
-        );
+        const { isMobile } = useWindowSize();
+        const itemSize = computed(() => !isMobile.value ? 72 : 68); // 64px + 4px margin between items
```
Comment (danimoh) [inline src/components/TransactionList.vue]:
```suggestion
        const itemSize = computed(() => isMobile.value ? 68 : 72); // mobile: 64px + 4px margin between items
```

File: src/components/modals/SendModal.vue [hunk 3]
```diff
@@ -558,11 +558,11 @@ export default defineComponent({
         const amountInputRef: Ref<AmountInput | null> = ref(null);
         const messageInputRef: Ref<LabelInput | null> = ref(null);
 
-        const { width } = useWindowSize();
+        const { isMobile } = useWindowSize();
 
         async function focus(elementRef: Ref<AddressInput | LabelInput | AmountInput | null>) {
             // TODO: Detect onscreen keyboards instead?
-            if (width.value <= 700) return; // Full mobile breakpoint
+            if (isMobile.value) return; // Full mobile breakpoint
```
Comment (danimoh) [inline src/components/modals/SendModal.vue]:
```suggestion
            if (isMobile.value) return;
```

File: src/composables/useWindowSize.ts [hunk 4]
```diff
@@ -8,17 +8,24 @@ let numberOfListeners = 0;
...
 export function useWindowSize() {
     // First-time setup
-    if (!width || !height) {
+    if (!width || !height || !isMobile || !isTablet || !isFullDesktop) {
         width = ref(0);
         height = ref(0);
         listener();
+        isMobile = computed(() => width!.value <= 700); // Full mobile breakpoint
+        isTablet = computed(() => width!.value <= 960); // Tablet breakpoint
+        isFullDesktop = computed(() => width!.value > 1160); // Half mobile breakpoint
```
Comment (danimoh) [inline src/composables/useWindowSize.ts]:
```suggestion
        isFullDesktop = computed(() => width!.value > 1160); // Desktop breakpoint
```

**Review by danimoh — APPROVED** [review]

**Review by onmax — COMMENTED** [review]
Looks good to me. As @danimoh said, once merged to #72 we have to update the PR

**Pull request merged**